application/container-builder/server/server.go

```diff
@@ -46,25 +50,48 @@ func NewServer() (*Server, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to create builder")
 	}
+	repoDirs := GetRepoDirOrDefault()
+	// create the repo directory if it doesn't exist
+	if _, err := os.Stat(repoDirs); os.IsNotExist(err) {
+		err = os.Mkdir(repoDirs, 0755)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to create repo directory")
+		}
+	}
+
 	l := zerolog.New(os.Stdout).With().Timestamp().Str("component", "server").Logger()
 	return &Server{
-		builder: builder,
-		log:     l,
+		builder:  builder,
+		repoDirs: repoDirs,
+		log:      l,
 	}, nil
 
 }
 
-// CreateContainer maps to the CreateContainer RPC
+// CreateContainer maps to the CreateContainer RPC``
+// call using grpcurl:
+// grpcurl -plaintext -d '{"repoURL": "your_repo_url", "type": "your_type", "customerID": "your_customer_id"}' localhost:4040 container.ContainerService/CreateContainer
+
 func (s *Server) CreateContainer(ctx context.Context, req *container.CreateContainerRequest) (*container.CreateContainerReply, error) {
+	s.log.Info().Msg("processing build for repo " + req.RepoURL)
 	buildID := uuid.New().String()
 	// create a build request in the db
-	repo := strings.TrimPrefix(req.RepoURL, "https://github.com")
-	err := s.builder.NewBuild(buildID, repo, req.CustomerID)
+	repo := strings.TrimPrefix(req.RepoURL, "https://github.com/")
+	err := s.builder.NewBuild(req.CustomerID, repo, buildID)
 	if err != nil {
 		s.log.Error().Err(err).Msg("unable to create build request")
 		return nil, errors.Wrap(err, "unable to create build , please try again")
 	}
 
+	dir := path.Join(s.repoDirs, buildID)
```

Comment:
@Klaven  how do you think it's best to name the directory? right now it's just the buildID?

Reply:
seems fine for now.